### PR TITLE
Education premium users now also have a dedicated Discord channel

### PR DESCRIPTION
**Pull request by ClementPasteau**

### CODE REVIEW

**Review by 4ian — APPROVED** [review]

File: newIDE/app/src/Profile/ProfileDetails.js [hunk 1]
```diff
@@ -257,11 +257,13 @@ const ProfileDetails = ({
...
-  const hasGoldOrProSubscription =
+  const canBenefitFromDiscordRole =
     !!subscription &&
     !!subscription.planId &&
-    (subscription.planId === 'gdevelop_gold' ||
-      subscription.planId === 'gdevelop_startup');
+    ['gdevelop_education', 'gdevelop_startup', 'gdevelop_gold'].includes(
+      subscription.planId
+    ) &&
+    !subscription.benefitsFromEducationPlan;
```
Comment (AlexandreSi) [inline newIDE/app/src/Profile/ProfileDetails.js:266]: 👍

File: newIDE/app/src/Profile/Subscription/SubscriptionPendingDialog.js [hunk 2]
```diff
@@ -30,11 +30,14 @@ export default function SubscriptionPendingDialog({
     !!authenticatedUser &&
     !!authenticatedUser.subscription &&
     !!authenticatedUser.subscription.planId;
-  const hasGoldOrStartupPlan =
+  const canBenefitFromDiscordRole =
```
Comment (AlexandreSi) [inline newIDE/app/src/Profile/Subscription/SubscriptionPendingDialog.js:33]: I reckon this should be a method shared between the 2 files for DRY purposes

File: newIDE/app/src/Profile/ProfileDetails.js (hunk 1 above)
Comment (AlexandreSi) [reply newIDE/app/src/Profile/ProfileDetails.js:266]: This is highly needed 😅

File: newIDE/app/src/Profile/Subscription/SubscriptionPendingDialog.js (hunk 2 above)
Comment (ClementPasteau) [reply newIDE/app/src/Profile/Subscription/SubscriptionPendingDialog.js:33]: yep, you're right I'll factor that

File: newIDE/app/src/Profile/ProfileDetails.js (hunk 1 above)
Comment (ClementPasteau) [reply newIDE/app/src/Profile/ProfileDetails.js:266]: Yes 😁

File: newIDE/app/src/Utils/GDevelopServices/Usage.spec.js
Comment (AlexandreSi) [inline newIDE/app/src/Utils/GDevelopServices/Usage.spec.js:1]: 💜

**Review by AlexandreSi — APPROVED** [review]

**Pull request merged**